services/wfm/src/main/java/org/openkilda/wfm/topology/flow/validation/FlowValidator.java

```diff
@@ -113,4 +122,41 @@ void checkFlowForEndpointConflicts(Flow requestedFlow) throws FlowValidationExce
                             conflictedFlow.get().getFlowId()));
         }
     }
+
+    /**
+     * Validates switch existing.
+     *
+     * @param requestedFlow a flow to be validated.
+     * @param switches a list of all switches.
+     * @throws SwitchValidationException if switch not found.
+     */
+    @SuppressWarnings("squid:S1192")
+    public void validateSwitches(Flow requestedFlow, List<SwitchInfoData> switches) throws SwitchValidationException {
+        final SwitchId sourceId = requestedFlow.getSourceSwitch();
+        final SwitchId destinationId = requestedFlow.getDestinationSwitch();
+
+        List<SwitchInfoData> foundSwitches = switches.stream().filter(switchData ->
+                Objects.equals(switchData.getSwitchId(), sourceId)
+                        || Objects.equals(switchData.getSwitchId(), destinationId))
+                .collect(Collectors.toList());
+        boolean source = foundSwitches.stream().anyMatch(sw -> Objects.equals(sw.getSwitchId(), sourceId));
+        boolean destination = foundSwitches.stream().anyMatch(sw -> Objects.equals(sw.getSwitchId(), destinationId));
+
+        if (!source && !destination) {
+            String errorMessage =
+                    String.format("Source switch %s and Destination switch %s are not connected to the controller",
+                            sourceId, destinationId);
+            logger.error(errorMessage);
```

Comment:
Put error in log(or any other place) should caller, only it know how this error should be handled.